src/game.cpp

```diff
@@ -2228,7 +2264,13 @@ input_context game::get_player_input(std::string &action)
         }
         inp_mngr.set_timeout(-1);
     } else {
-        while (handle_mouseview(ctxt, action)) {};
+        inp_mngr.set_timeout(125);
+        while (handle_mouseview(ctxt, action)) {
+            if (action == "TIMEOUT" && has_user_turn_timeout_elapsed()) {
```

Comment:
Why did you check for turn timeout again here? It should only be set to timeout if a previous check already returned true.